# #Dataset Overview

*Number of observations (n): 2938

*Number of predictors (p) :21

n≥p+5⇒2938≥21+5   
so the data is suitable for multivariable regression

# #features explaination
###Demographic & Identifiers

*   Country
*   year: Year of observation (2000–2015)
*   Status: Developing or Developed country.

###Health & Mortality Indicators

*   Adult Mortality:Probability of dying between ages 15–60 per 1000 population.
*   infant deaths: Number of infant deaths per 1000 live births.
*   under-five deaths: Deaths of children under 5 per 1000 population.
*   Hepatitis B : Immunization coverage against the Hepatitis B.
*   Polio : Polio immunization rate.
*   Diphtheria: DPT vaccination coverage.
*   Measles: number of reported cases per 1000 population.
*   HIV/AIDS: Deaths per 1000 population.

##Lifestyle & Physical Health


*   Alcohol:The average annual alcohol consumption per individual.
*   BMI:Average Body Mass Index.  
*   thinness 1–19 years: Malnutrition indicator for youth.
*   thinness 5–9 years: Malnutrition indicator for children.

##Economic Indicators
*   GDP:The average income generated per person in a country
*   percentage expenditure:Health expenditure as a percentage of GDP.
*   Total expenditure:Government health spending (%).   
*  Income composition of resources:Composite income index; strong predictor of development.

##Education & Social Development

*  Schooling : Average years of schooling
*  Population : The total population size of a country



In [271]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

# Load the dataset

In [272]:
data = pd.read_csv('/content/Life Expectancy Data.csv')

In [273]:
data

Country  Year      Status  Life expectancy   Adult Mortality  \
0     Afghanistan  2015  Developing              65.0            263.0   
1     Afghanistan  2014  Developing              59.9            271.0   
2     Afghanistan  2013  Developing              59.9            268.0   
3     Afghanistan  2012  Developing              59.5            272.0   
4     Afghanistan  2011  Developing              59.2            275.0   
...           ...   ...         ...               ...              ...   
2933     Zimbabwe  2004  Developing              44.3            723.0   
2934     Zimbabwe  2003  Developing              44.5            715.0   
2935     Zimbabwe  2002  Developing              44.8             73.0   
2936     Zimbabwe  2001  Developing              45.3            686.0   
2937     Zimbabwe  2000  Developing              46.0            665.0   

      infant deaths  Alcohol  percentage expenditure  Hepatitis B  Measles   \
0                62     0.01               71.279624         65.0      1154   
1                64     0.01               73.523582         62.0       492   
2                66     0.01               73.219243         64.0       430   
3                69     0.01               78.184215         67.0      2787   
4                71     0.01                7.097109         68.0      3013   
...             ...      ...                     ...          ...       ...   
2933             27     4.36                0.000000         68.0        31   
2934             26     4.06                0.000000          7.0       998   
2935             25     4.43                0.000000         73.0       304   
2936             25     1.72                0.000000         76.0       529   
2937             24     1.68                0.000000         79.0      1483   

      ...  Polio  Total expenditure  Diphtheria    HIV/AIDS         GDP  \
0     ...    6.0               8.16         65.0        0.1  584.259210   
1     ...   58.0               8.18         62.0        0.1  612.696514   
2     ...   62.0               8.13         64.0        0.1  631.744976   
3     ...   67.0               8.52         67.0        0.1  669.959000   
4     ...   68.0               7.87         68.0        0.1   63.537231   
...   ...    ...                ...          ...        ...         ...   
2933  ...   67.0               7.13         65.0       33.6  454.366654   
2934  ...    7.0               6.52         68.0       36.7  453.351155   
2935  ...   73.0               6.53         71.0       39.8   57.348340   
2936  ...   76.0               6.16         75.0       42.1  548.587312   
2937  ...   78.0               7.10         78.0       43.5  547.358878   

      Population   thinness  1-19 years   thinness 5-9 years  \
0     33736494.0                   17.2                 17.3   
1       327582.0                   17.5                 17.5   
2     31731688.0                   17.7                 17.7   
3      3696958.0                   17.9                 18.0   
4      2978599.0                   18.2                 18.2   
...          ...                    ...                  ...   
2933  12777511.0                    9.4                  9.4   
2934  12633897.0                    9.8                  9.9   
2935    125525.0                    1.2                  1.3   
2936  12366165.0                    1.6                  1.7   
2937  12222251.0                   11.0                 11.2   

      Income composition of resources  Schooling  
0                               0.479       10.1  
1                               0.476       10.0  
2                               0.470        9.9  
3                               0.463        9.8  
4                               0.454        9.5  
...                               ...        ...  
2933                            0.407        9.2  
2934                            0.418        9.5  
2935                            0.427      

# Check the dimension of the dataset

In [274]:
data.shape

(2938, 22)

2938 rows → number of observations (n)

22 columns → number of variables (features + target)



# Get information about the data

In [275]:
# Display column names and data types
data.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2938 entries, 0 to 2937
Data columns (total 22 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   Country                          2938 non-null   object 
 1   Year                             2938 non-null   int64  
 2   Status                           2938 non-null   object 
 3   Life expectancy                  2928 non-null   float64
 4   Adult Mortality                  2928 non-null   float64
 5   infant deaths                    2938 non-null   int64  
 6   Alcohol                          2744 non-null   float64
 7   percentage expenditure           2938 non-null   float64
 8   Hepatitis B                      2385 non-null   float64
 9   Measles                          2938 non-null   int64  
 10   BMI                             2904 non-null   float64
 11  under-five deaths                2938 non-null   int64  
 12  Polio               

# Classification of Variables in the Dataset


In [276]:
# Categorical variables
categorical_variables = ["Country", "Status"]

# Numerical variables
numerical_variables = data.columns.drop(categorical_variables)

numerical_variables

Index(['Year', 'Life expectancy ', 'Adult Mortality', 'infant deaths',
       'Alcohol', 'percentage expenditure', 'Hepatitis B', 'Measles ', ' BMI ',
       'under-five deaths ', 'Polio', 'Total expenditure', 'Diphtheria ',
       ' HIV/AIDS', 'GDP', 'Population', ' thinness  1-19 years',
       ' thinness 5-9 years', 'Income composition of resources', 'Schooling'],
      dtype='object')

In [277]:
data["Country"].unique()

array(['Afghanistan', 'Albania', 'Algeria', 'Angola',
       'Antigua and Barbuda', 'Argentina', 'Armenia', 'Australia',
       'Austria', 'Azerbaijan', 'Bahamas', 'Bahrain', 'Bangladesh',
       'Barbados', 'Belarus', 'Belgium', 'Belize', 'Benin', 'Bhutan',
       'Bolivia (Plurinational State of)', 'Bosnia and Herzegovina',
       'Botswana', 'Brazil', 'Brunei Darussalam', 'Bulgaria',
       'Burkina Faso', 'Burundi', "Côte d'Ivoire", 'Cabo Verde',
       'Cambodia', 'Cameroon', 'Canada', 'Central African Republic',
       'Chad', 'Chile', 'China', 'Colombia', 'Comoros', 'Congo',
       'Cook Islands', 'Costa Rica', 'Croatia', 'Cuba', 'Cyprus',
       'Czechia', "Democratic People's Republic of Korea",
       'Democratic Republic of the Congo', 'Denmark', 'Djibouti',
       'Dominica', 'Dominican Republic', 'Ecuador', 'Egypt',
       'El Salvador', 'Equatorial Guinea', 'Eritrea', 'Estonia',
       'Ethiopia', 'Fiji', 'Finland', 'France', 'Gabon', 'Gambia',
       'Georgia', 'Germany'

In [278]:
data["Country"].unique().size

193

Country is a nominal categorical variable → have 193 unique categories, would create hundreds of dummy variables so we will drop this column instead of making one hot encoding and the country does not affect the life expectancy

Status is a binary categorical variable →Label Encoding

```
Developed → 1, Developing → 0
```






# Encoding for categorical features

In [279]:
# Clean Status column first to avoid missing value problem in status after encoding
data["Status"] = data["Status"].str.strip()

# Encode Status
data["Status"] = data["Status"].map({"Developed": 1, "Developing": 0})

# Drop Country column
data.drop(columns=["Country"], inplace=True)



In [280]:
data['Status']

0       0
1       0
2       0
3       0
4       0
       ..
2933    0
2934    0
2935    0
2936    0
2937    0
Name: Status, Length: 2938, dtype: int64

# Verify encoding

In [281]:
data.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 2938 entries, 0 to 2937
Data columns (total 21 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   Year                             2938 non-null   int64  
 1   Status                           2938 non-null   int64  
 2   Life expectancy                  2928 non-null   float64
 3   Adult Mortality                  2928 non-null   float64
 4   infant deaths                    2938 non-null   int64  
 5   Alcohol                          2744 non-null   float64
 6   percentage expenditure           2938 non-null   float64
 7   Hepatitis B                      2385 non-null   float64
 8   Measles                          2938 non-null   int64  
 9    BMI                             2904 non-null   float64
 10  under-five deaths                2938 non-null   int64  
 11  Polio                            2919 non-null   float64
 12  Total expenditure   

# Count missing values for each column


In [282]:
missing_count = data.isnull().sum()
missing_count

Year                                 0
Status                               0
Life expectancy                     10
Adult Mortality                     10
infant deaths                        0
Alcohol                            194
percentage expenditure               0
Hepatitis B                        553
Measles                              0
 BMI                                34
under-five deaths                    0
Polio                               19
Total expenditure                  226
Diphtheria                          19
 HIV/AIDS                            0
GDP                                448
Population                         652
 thinness  1-19 years               34
 thinness 5-9 years                 34
Income composition of resources    167
Schooling                          163
dtype: int64

# Drop the missing values in target variable ['Life expectancy']

In [283]:
data = data.dropna(subset=['Life expectancy '])

#Immunization columns (Hepatitis B – Polio – Diphtheria)
 Missing values in immunization variables were imputed using group-wise median
 based on country development status to preserve realistic differences
 between developed and developing countries.


In [284]:
# Group-wise median imputation for immunization variables
immunization_cols = ["Hepatitis B", "Polio", "Diphtheria "]

for col in immunization_cols:
    data[col] = data.groupby("Status")[col].transform(
        lambda x: x.fillna(x.median())
    )

/tmp/ipython-input-381129126.py:5: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data[col] = data.groupby("Status")[col].transform(


#Social Variables (Schooling – Income composition)
Social variables are imputed using group-wise median by development status
 because education and income composition are strongly associated with a
 country's level of development.

In [285]:
social_cols = ["Schooling", "Income composition of resources"]

for col in social_cols:
    data[col] = data.groupby("Status")[col].transform(
        lambda x: x.fillna(x.median())
    )


/tmp/ipython-input-3470988948.py:4: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data[col] = data.groupby("Status")[col].transform(


#Economic variables
GDP and Population are imputed using group-wise median due to strong
 association with development status.

Total health expenditure is imputed using the global median.

In [286]:
# GDP and Population are imputed using group-wise median
for col in ["GDP", "Population"]:
    data[col] = data.groupby("Status")[col].transform(
        lambda x: x.fillna(x.median())
    )

# Total health expenditure is imputed using the global median.
data["Total expenditure"] = data["Total expenditure"].fillna(
    data["Total expenditure"].median()
)


/tmp/ipython-input-3436466739.py:3: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data[col] = data.groupby("Status")[col].transform(
/tmp/ipython-input-3436466739.py:8: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data["Total expenditure"] = data["Total expenditure"].fillna(


## Lifestyle-related variables are imputed using the median of each column.

In [287]:
lifestyle_cols = ["Alcohol",' BMI '," thinness  1-19 years", " thinness 5-9 years"]

# Impute lifestyle variables using median
for col in lifestyle_cols:
    data.loc[:, col] = data[col].fillna(data[col].median())

#check that no missing values remain in the dataset

In [288]:
data.isnull().sum()

Year                               0
Status                             0
Life expectancy                    0
Adult Mortality                    0
infant deaths                      0
Alcohol                            0
percentage expenditure             0
Hepatitis B                        0
Measles                            0
 BMI                               0
under-five deaths                  0
Polio                              0
Total expenditure                  0
Diphtheria                         0
 HIV/AIDS                          0
GDP                                0
Population                         0
 thinness  1-19 years              0
 thinness 5-9 years                0
Income composition of resources    0
Schooling                          0
dtype: int64

## Outlier Detection using IQR

In [289]:
outlier_summary = {}  # Dictionary to store the number of outliers for each column

for col in numerical_variables:
    Q1 = data[col].quantile(0.25)  # 25th percentile
    Q3 = data[col].quantile(0.75)  # 75th percentile
    IQR = Q3 - Q1                  # Interquartile range (IQR)

    lower_bound = Q1 - 1.5 * IQR   # Lower bound
    upper_bound = Q3 + 1.5 * IQR   # Upper bound

    # Select rows with value is below lower_bound or above upper_bound
    outliers = data[(data[col] < lower_bound) | (data[col] > upper_bound)]

    outlier_summary[col] = outliers.shape[0]  # Store the count of outliers

# Convert the dictionary to a DataFrame
pd.DataFrame.from_dict(outlier_summary, orient="index", columns=["Outlier Count"])


Outlier Count
Year                                         0
Life expectancy                             10
Adult Mortality                             82
infant deaths                              315
Alcohol                                      3
percentage expenditure                     388
Hepatitis B                                320
Measles                                    542
 BMI                                         0
under-five deaths                          394
Polio                                      278
Total expenditure                           49
Diphtheria                                 297
 HIV/AIDS                                  542
GDP                                        459
Population                                 452
 thinness  1-19 years                      100
 thinness 5-9 years                         99
Income composition of resources            130
Schooling                                   59

#Outliers Capping
 ### Cap values below the lower bound to the lower bound
 ### Cap values above the upper bound to the upper bound

In [290]:
for col in numerical_variables:
    Q1 = data[col].quantile(0.25)
    Q3 = data[col].quantile(0.75)
    IQR = Q3 - Q1

    lower_bound = Q1 - 1.5 * IQR  # Lower bound for capping
    upper_bound = Q3 + 1.5 * IQR  # Upper bound for capping

    # Capping
    data[col] = np.where(data[col] < lower_bound, lower_bound, data[col])
    data[col] = np.where(data[col] > upper_bound, upper_bound, data[col])


/tmp/ipython-input-33854993.py:10: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data[col] = np.where(data[col] < lower_bound, lower_bound, data[col])
/tmp/ipython-input-33854993.py:11: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  data[col] = np.where(data[col] > upper_bound, upper_bound, data[col])
/tmp/ipython-input-33854993.py:10: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the doc

#Check for remaining outliers after capping


In [291]:
outlier_summary_after = {}

for col in numerical_variables:
    Q1 = data[col].quantile(0.25)
    Q3 = data[col].quantile(0.75)
    IQR = Q3 - Q1
    lower_bound = Q1 - 1.5 * IQR
    upper_bound = Q3 + 1.5 * IQR

    outliers = data[(data[col] < lower_bound) | (data[col] > upper_bound)]
    outlier_summary_after[col] = outliers.shape[0]

pd.DataFrame.from_dict(outlier_summary_after, orient="index", columns=["Outlier Count"])


Outlier Count
Year                                         0
Life expectancy                              0
Adult Mortality                              0
infant deaths                                0
Alcohol                                      0
percentage expenditure                       0
Hepatitis B                                  0
Measles                                      0
 BMI                                         0
under-five deaths                            0
Polio                                        0
Total expenditure                            0
Diphtheria                                   0
 HIV/AIDS                                    0
GDP                                          0
Population                                   0
 thinness  1-19 years                        0
 thinness 5-9 years                          0
Income composition of resources              0
Schooling                                    0

#Descriptive Statistics

In [292]:
data.describe()

Year       Status  Life expectancy   Adult Mortality  \
count  2928.00000  2928.000000       2928.000000      2928.000000   
mean   2007.50000     0.174863         69.233197       162.087432   
std       4.61056     0.379915          9.500077       115.866689   
min    2000.00000     0.000000         44.200000         1.000000   
25%    2003.75000     0.000000         63.100000        74.000000   
50%    2007.50000     0.000000         72.100000       144.000000   
75%    2011.25000     0.000000         75.700000       228.000000   
max    2015.00000     1.000000         89.000000       459.000000   

       infant deaths      Alcohol  percentage expenditure  Hepatitis B  \
count    2928.000000  2928.000000             2928.000000  2928.000000   
mean       13.681694     4.558602              284.889600    87.057719   
std        19.124887     3.918707              390.385763    12.342527   
min         0.000000     0.010000                0.000000    61.000000   
25%         0.000000     1.107500                4.853964    82.000000   
50%         3.000000     3.770000               65.611455    91.000000   
75%        22.000000     7.400000              442.614322    96.000000   
max        55.000000    16.838750             1099.254858    99.000000   

          Measles          BMI   ...        Polio  Total expenditure  \
count  2928.000000  2928.000000  ...  2928.000000        2928.000000   
mean    235.469518    38.291291  ...    85.337432           5.886021   
std     355.694920    19.857308  ...    15.803728           2.286799   
min       0.000000     1.000000  ...    49.500000           0.370000   
25%       0.000000    19.400000  ...    78.000000           4.370000   
50%      17.000000    43.350000  ...    93.000000           5.750000   
75%     362.250000    56.100000  ...    97.000000           7.330000   
max     905.625000    77.600000  ...    99.000000          11.770000   

       Diphtheria      HIV/AIDS           GDP    Population  \
count  2928.000000  2928.000000   2928.000000  2.928000e+03   
mean     85.282616     0.537705   3863.743924  3.181501e+06   
std      15.784674     0.694107   4510.057125  3.850164e+06   
min      49.500000     0.100000      1.681350  3.400000e+01   
25%      78.000000     0.100000    578.797095  4.181205e+05   
50%      93.000000     0.100000   1305.304687  1.412966e+06   
75%      97.000000     0.800000   5453.282630  4.592777e+06   
max      99.000000     1.850000  12765.010932  1.085476e+07   

        thinness  1-19 years   thinness 5-9 years  \
count            2928.000000          2928.000000   
mean                4.698805             4.718545   
std                 3.933940             3.976766   
min                 0.100000             0.100000   
25%                 1.600000             1.600000   
50%                 3.300000             3.400000   
75%                 7.100000             7.200000   
max                15.350000            15.600000   

       Income composition of resources    Schooling  
count                      2928.000000  2928.000000  
mean                          0.635381    12.094638  
std                           0.195652     3.154904  
min                           0.089625     4.300000  
25%                           0.504000    10.300000  
50%                           0.673000    12.300000  
75%                           0.780250    14.300000  
max                           0.948000    20.300000  

[8 rows x 21 columns]

## Measures of Central Tendency


In [293]:
central_tendency = data[numerical_variables].agg(["mean", "median"])
central_tendency


Year  Life expectancy   Adult Mortality  infant deaths   Alcohol  \
mean    2007.5         69.233197       162.087432      13.681694  4.558602   
median  2007.5         72.100000       144.000000       3.000000  3.770000   

        percentage expenditure  Hepatitis B    Measles        BMI   \
mean                284.889600    87.057719  235.469518  38.291291   
median               65.611455    91.000000   17.000000  43.350000   

        under-five deaths       Polio  Total expenditure  Diphtheria   \
mean             18.027322  85.337432           5.886021    85.282616   
median            4.000000  93.000000           5.750000    93.000000   

         HIV/AIDS          GDP    Population   thinness  1-19 years  \
mean     0.537705  3863.743924  3.181501e+06               4.698805   
median   0.100000  1305.304687  1.412966e+06               3.300000   

         thinness 5-9 years  Income composition of resources  Schooling  
mean               4.718545                         0.635381  12.094638  
median             3.400000                         0.673000  12.300000

In [294]:
dispersion = data[numerical_variables].agg(["std", "var", "min", "max"])
dispersion


Year  Life expectancy   Adult Mortality  infant deaths    Alcohol  \
std     4.61056          9.500077       115.866689      19.124887   3.918707   
var    21.25726         90.251470     13425.089722     365.761305  15.356264   
min  2000.00000         44.200000         1.000000       0.000000   0.010000   
max  2015.00000         89.000000       459.000000      55.000000  16.838750   

     percentage expenditure  Hepatitis B       Measles         BMI   \
std              390.385763    12.342527     355.694920   19.857308   
var           152401.043890   152.337973  126518.876409  394.312678   
min                0.000000    61.000000       0.000000    1.000000   
max             1099.254858    99.000000     905.625000   77.600000   

     under-five deaths        Polio  Total expenditure  Diphtheria   \
std           25.215584   15.803728           2.286799    15.784674   
var          635.825697  249.757813           5.229448   249.155923   
min            0.000000   49.500000           0.370000    49.500000   
max           70.000000   99.000000          11.770000    99.000000   

      HIV/AIDS           GDP    Population   thinness  1-19 years  \
std   0.694107  4.510057e+03  3.850164e+06               3.933940   
var   0.481784  2.034062e+07  1.482376e+13              15.475885   
min   0.100000  1.681350e+00  3.400000e+01               0.100000   
max   1.850000  1.276501e+04  1.085476e+07              15.350000   

      thinness 5-9 years  Income composition of resources  Schooling  
std             3.976766                         0.195652   3.154904  
var            15.814671                         0.038280   9.953422  
min             0.100000                         0.089625   4.300000  
max            15.600000                         0.948000  20.300000

#Split Data to Target & Features for regreesion model

In [295]:
# Define target variable
y = data['Life expectancy ']

# Define feature matrix (drop target)
X = data.drop(columns=['Life expectancy '])


# Compute descriptive statistics to the features



In [296]:
feature_means = X.mean()
feature_stds = X.std()

#Feature Standardization (Z-score)

In [297]:
X_standardized = (X - feature_means) / feature_stds
X_standardized.head()

Year   Status  Adult Mortality  infant deaths   Alcohol  \
0  1.626701 -0.46027         0.870937       2.160447 -1.160741   
1  1.409807 -0.46027         0.939982       2.160447 -1.160741   
2  1.192914 -0.46027         0.914090       2.160447 -1.160741   
3  0.976020 -0.46027         0.948612       2.160447 -1.160741   
4  0.759127 -0.46027         0.974504       2.160447 -1.160741   

   percentage expenditure  Hepatitis B  Measles       BMI   \
0               -0.547177    -1.787131  1.884074 -0.966460   
1               -0.541429    -2.030194  0.721209 -0.991640   
2               -0.542208    -1.868152  0.546903 -1.016819   
3               -0.529490    -1.625090  1.884074 -1.041999   
4               -0.711585    -1.544069  1.884074 -1.062143   

   under-five deaths      Polio  Total expenditure  Diphtheria    HIV/AIDS  \
0            2.061133 -2.267657           0.994394    -1.284956  -0.630602   
1            2.061133 -1.729809           1.003140    -1.475014  -0.630602   
2            2.061133 -1.476704           0.981275    -1.348309  -0.630602   
3            2.061133 -1.160323           1.151819    -1.158251  -0.630602   
4            2.061133 -1.097047           0.867579    -1.094899  -0.630602   

        GDP  Population   thinness  1-19 years   thinness 5-9 years  \
0 -0.727149    1.992970               2.707513             2.736257   
1 -0.720844   -0.741246               2.707513             2.736257   
2 -0.716620    1.992970               2.707513             2.736257   
3 -0.708147    0.133879               2.707513             2.736257   
4 -0.842607   -0.052700               2.707513             2.736257   

   Income composition of resources  Schooling  
0                        -0.799278  -0.632234  
1                        -0.814611  -0.663931  
2                        -0.845278  -0.695627  
3                        -0.881056  -0.727324  
4                        -0.927056  -0.822414